src/internal.c

```diff
@@ -15830,25 +15824,15 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                 #endif
 #if defined(HAVE_CERTIFICATE_STATUS_REQUEST_V2)
                     if (ret == 0 && addToPendingCAs && !alreadySigner) {
-#ifdef WOLFSSL_SMALL_STACK
-                        DecodedCert *dCertAdd = NULL;
-#else
-                        DecodedCert dCertAdd[1];
-#endif
+                        WC_DECLARE_VAR(dCertAdd, DecodedCert, 1, 0);
                         int dCertAdd_inited = 0;
                         DerBuffer *derBuffer = NULL;
                         buffer* cert = &args->certs[args->certIdx];
                         Signer *s = NULL;
 
-#ifdef WOLFSSL_SMALL_STACK
-                        dCertAdd = (DecodedCert *)
-                            XMALLOC(sizeof(*dCertAdd), ssl->heap,
-                                    DYNAMIC_TYPE_TMP_BUFFER);
-                        if (dCertAdd == NULL) {
-                            ret = MEMORY_E;
-                            goto exit_req_v2;
-                        }
-#endif
+                        WC_ALLOC_VAR_EX(dCertAdd, DecodedCert, 1, ssl->heap,
+                            DYNAMIC_TYPE_TMP_BUFFER,
+                            {ret=MEMORY_E;goto exit_req_v2;});
```

Comment:
Any ideas for improving the readability of this? `{ret=MEMORY_E;goto exit_req_v2;});`. Its acceptable as-is, just curious if you have an idea to improve it.